.github/workflows/ci.yaml

```diff
@@ -16,7 +16,7 @@ jobs:
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        java: [8, 11, 17, 21, 23]
+        java: [8, 11, 17, 21, 24]
         # macos-13 is x86, macos-latest is aarch64
         os: [ubuntu-latest, macos-13, macos-latest]
```

Comment:
While we're updating the matrix, a few things:
1. We should consider adding [ubuntu-24.04-arm](https://github.com/actions/partner-runner-images/blob/main/images/arm-ubuntu-24-image.md) to get ARM coverage on Linux.  (The macos-latest gets Apple Silicon while -13 is x86.) See https://docs.github.com/en/actions/using-github-hosted-runners/using-github-hosted-runners/about-github-hosted-runners#standard-github-hosted-runners-for-public-repositories 
2. We might consider defining some combinations to cover all JDKs and all OS platforms without doing every possible combination (4 x 5 = 20).  All OS's combos on the latest JDK seems sane but maybe spread the 4 other JDKs over 4 os's for a total of 8 runs?

Reply:
How about:

- java: [8, 21, 24]
- os: [ubuntu-latest, ubuntu-24.04-arm, macos-latest, maxos-13]

Idea here:

- cover the lower end of the supported spectrum (JDK 8)
- cover the current LTS
- cover the current Java version

At this point in time, I don't see many things breaking on JDK 11/17, that will work both on JDK 8 _and_ 21.

Reply:
Yep. That makes sense.